pandas/core/dtypes/dtypes.py

```diff
@@ -338,36 +329,45 @@ def _from_values_or_dtype(
 
         return dtype
 
+    @classmethod
+    def construct_from_string(cls, string: str):
```

Comment:
Can add a docstring here if needed.  For Python 3.5+, [`inspect.getdoc`](https://docs.python.org/3/library/inspect.html#inspect.getdoc) will inherit from the parent class if docstrings aren't overridden.  Basically this means that `CategoricalDtype.construct_from_string.__doc__` will be `None` but doing things like `help(CategoricalDtype.construct_from_string)` or `CategoricalDtype.construct_from_string?` will show the docstring from the parent class.  Not sure if we have other reasons for needing the `__doc__` attribute to exist though.